gobblin-core/src/main/java/org/apache/gobblin/converter/parquet/JsonElementConversionFactory.java

```diff
@@ -0,0 +1,695 @@
+    private JsonElementConverter elementConverter;
+    private PrimitiveTypeName elementPrimitiveName;
+    private SourceType elementSourceType;
+
+    public ComplexConverterForUniformElementTypes(String fieldName, boolean nullable, JsonObject schemaNode,
+        String itemKey, boolean repeated) {
+      super(fieldName, nullable, repeated);
+      this.elementPrimitiveName = getPrimitiveTypeInParquet(schemaNode, itemKey);
+      this.elementSourceType = getPrimitiveTypeInSource(schemaNode, itemKey);
+    }
+
+    /**
+     * Set a {@link JsonElementConverter} for the elements present within Complex data types.
+     * @param elementConverter
+     */
+    protected void setElementConverter(JsonElementConverter elementConverter) {
+      this.elementConverter = elementConverter;
+    }
+
+    public JsonElementConverter getElementConverter() {
+      return this.elementConverter;
+    }
+
+    @Override
+    public PrimitiveTypeName getTargetType() {
+      throw new UnsupportedOperationException("Complex types does not support PrimitiveTypeName");
+    }
+
+    /**
+     * {@link PrimitiveTypeName} of the elements composed within complex type.
+     * @param schemaNode
+     * @param itemKey
+     * @return
+     */
+    private PrimitiveTypeName getPrimitiveTypeInParquet(JsonObject schemaNode, String itemKey) {
+      return typeMap.get(getPrimitiveTypeInSource(schemaNode, itemKey));
+    }
+
+    /**
+     * {@link SourceType} of the elements composed within complex type.
+     * @param schemaNode
+     * @param itemKey
+     * @return
+     */
+    private SourceType getPrimitiveTypeInSource(JsonObject schemaNode, String itemKey) {
+      String type = schemaNode.get("dataType").getAsJsonObject().get(itemKey).getAsString().toUpperCase();
+      return SourceType.valueOf(type);
+    }
+
+    public PrimitiveTypeName getElementTypeParquet() {
+      return elementPrimitiveName;
+    }
+
+    public SourceType getElementTypeSource() {
+      return elementSourceType;
+    }
+
+    /**
+     * Prepare a temp schema object of composed elements for passing to element converters.
+     * @return
+     */
+    protected JsonObject getElementSchema() {
+      String typeOfElement = getElementTypeSource().toString();
+      JsonObject temp = new JsonObject();
+      JsonObject dataType = new JsonObject();
+      temp.addProperty("columnName", "temp");
+      dataType.addProperty("type", typeOfElement);
+      temp.add("dataType", dataType);
+      return temp;
+    }
+  }
+
+  public static class IntConverter extends JsonElementConverter {
+
+    public IntConverter(String fieldName, boolean nullable, boolean repeated) {
+      super(fieldName, nullable, repeated);
+    }
+
+    @Override
+    IntegerValue convertField(JsonElement value) {
+      return new IntegerValue(value.getAsInt());
+    }
+
+    @Override
+    public PrimitiveTypeName getTargetType() {
```

Comment:
When the `sourceType` is set during initialization, the `getTargetType` and `getSourceType` methods for the following simple converters can just use the default implementations.

Reply:
Realised sourceType is not required however targetType can be passed by specific Converters from constructor to the super. Example, IntConverter can pass INT32 to his super i.e JsonElementConverter